proforma.schema.json

```diff
@@ -0,0 +1,660 @@
+            ],
+            "description": "A region in a peptide sequence"
+        },
+        "sequence_ambiguous": {
+            "items": {
+                "$ref": "#/$defs/sequence_element"
+            },
+            "type": "array"
+        },
+        "sequence_element": {
+            "additionalProperties": false,
+            "properties": {
+                "amino_acid": {
+                    "$ref": "#/$defs/amino_acid"
+                },
+                "modifications": {
+                    "items": {
+                        "$ref": "#/$defs/modification"
+                    },
+                    "type": "array"
+                }
+            },
+            "required": [
+                "amino_acid"
+            ],
+            "type": "object"
+        },
+        "sequence_region": {
+            "additionalProperties": false,
+            "properties": {
+                "modifications": {
+                    "items": {
+                        "$ref": "#/$defs/modification"
+                    },
+                    "type": "array"
+                },
+                "sequence": {
+                    "items": {
+                        "$ref": "#/$defs/sequence_element"
+                    },
+                    "type": "array"
+                }
+            },
+            "required": [
+                "sequence",
+                "modifications"
+            ],
+            "type": "object"
+        },
+        "tag_accession": {
+            "additionalProperties": false,
+            "description": "The accession for a modification",
+            "properties": {
+                "accession": {
+                    "description": "The accession number, or for GNOme and RESID the string",
+                    "type": "string",
+                    "minLength": 1
+                },
+                "cv": {
+                    "$ref": "#/$defs/CV"
+                }
+            },
+            "required": [
+                "accession",
+                "cv"
+            ],
+            "type": "object"
+        },
+        "tag_formula": {
+            "$ref": "#/$defs/charged_formula",
+            "description": "A Formula:xxx modification"
+        },
+        "tag_glycan": {
+            "description": "A glycan composition modification",
+            "items": {
+                "additionalProperties": false,
+                "properties": {
+                    "monosaccharide": {
+                        "$ref": "#/$defs/monosaccharide"
+                    },
+                    "occurance": {
+                        "description": "The number of times this part occurs",
+                        "type": "integer"
+                    }
+                },
+                "required": [
+                    "monosaccharide",
+                    "occurance"
+                ],
+                "type": "object"
+            },
+            "type": "array"
+        },
+        "tag_info": {
```

Comment:
It is legal to have a tag that has both a value and an INFO component, see page 21 of the specification. These aren't separate tags. The same goes for the combined "synonymous" representation of modifications and/or masses on page 22.

It might be better to structure a `tag` as being composed of zero or more "tag_definitions", where the definitions are expressed as a `oneOf` like in the current `tags` formulation  as well as a label e.g. `#XL` and/or a localization score.

Reply:
Yes that is what i used the structure `modification` for. So currently the naming used is `tag_*` for any single item (CV/Name/Mass/etc) then `tags` for a collection of tags that all (should) define the same thing eg `Oxidation|Formula:O|Info:This is on a cool new place, never seen this before`. Then `modification` is the last thing containing tags and/or names, so essentially this is whatever someone can write between square brackets.

The naming is hard, if you have better names let me know.